examples/annotation-export/app.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path
+
+import matplotlib.dates as mdates
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+from shiny import App, Inputs, Outputs, Session, reactive, render, ui
+from shiny.plotutils import brushed_points
+
+app_ui = ui.page_fluid(
+    ui.panel_title("Plot annotation example"),
+    ui.p(
+        """
+        Select points to annotate them,
+        the plot is drawn with seaborn and all interaction is handled by Shiny.
+        """,
+        {"style": "font-size: larger"},
+    ),
+    ui.row(
+        ui.column(
+            6,
+            ui.output_plot("time_series", brush=ui.brush_opts(direction="x")),
+            ui.output_ui("annotator"),
+        ),
+        ui.column(
+            4,
+            ui.output_data_frame("annotations"),
+        ),
+        ui.column(2, ui.download_button("download", "Download CSV")),
+    ),
+)
+
+
+def server(input: Inputs, output: Outputs, session: Session):
+    annotation_values = reactive.Value(None)
+
+    @reactive.Calc
+    def ts_data():
+        path = Path(__file__).parent / "boulder_temp.csv"
+        dataframe = pd.read_csv(path)
+        dataframe["date"] = pd.to_datetime(dataframe["date"])
+        dataframe["annotation"] = ""
+
+        annotated = annotation_values.get()
+        if annotated is None:
+            out = dataframe
+        else:
+            out = annotated
+        return out
+
+    @reactive.Calc
+    def selected_data():
+        out = brushed_points(ts_data(), input.time_series_brush(), xvar="date")
+        return out
+
+    @reactive.Effect
+    @reactive.event(input.annotate_button)
+    def _():
+        selected = selected_data().copy()
+        selected["annotation_new"] = input.annotation()
+        selected = selected.loc[:, ["date", "annotation_new"]]
+
+        df = ts_data().copy()
+
+        df = df.merge(selected, on="date", how="left")
+        df["annotation_new"] = df["annotation_new"].fillna("")
+        updated_rows = df["annotation_new"] != ""
+        df.loc[updated_rows, "annotation"] = df.loc[updated_rows, "annotation_new"]
+        df = df.loc[:, ["date", "temp_c", "annotation"]]
+        annotation_values.set(df)
+
+    @output
+    @render.plot
+    def time_series():
+        fig, ax = plt.subplots()
+        ax.xaxis.set_major_locator(mdates.MonthLocator())
+        ax.xaxis.set_major_formatter(mdates.DateFormatter("%Y-%m"))
+        ax.set_title("Temperature readings, Boulder Colorado")
+        out = sns.scatterplot(
+            data=ts_data(), x="date", y="temp_c", hue="annotation", ax=ax
+        )
+
+        out.tick_params(axis="x", rotation=30)
+        return out.get_figure()
+
+    @output
+    @render.ui
+    def annotator():
+        if input.time_series_brush() is not None:
+            selected = selected_data().copy()
+
```

Comment:
I think this `copy()` isn't needed.

Now that I think about it, how about updating this app to use copy on write? That should eliminate the need for using `.copy()`.
https://github.com/rstudio/py-shiny/issues/614
https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html

Reply:
I tried a few things, and it turns out that moving some date time processing up in the script means that all of the copies can be removed without needing the pandas copy-on-modify.

Reply:
It looks like there's still a place where the `reactive.Value` `ts_data()` is being modified in place; the modified value is downloaded in `download`, where it calls `ts_data.to_csv()`. This results in the expected overall behavior for the app, but it relies on a dangerous modification-in-place. I'll make a fix.